python/bindings.cpp

```diff
@@ -51,6 +51,12 @@ py::array asArray(std::vector<std::string>&& values) {
     return py::array(py::dtype("object"), ptr->size(), ptr->data(), freeWhenDone(ptr));
 }
 
+template <typename T>
+py::array asArray(std::vector<T>&& values, ssize_t n_cols, ssize_t n_rows) {
+    auto ptr = new std::vector<T>(std::move(values));
+    return py::array({n_cols, n_rows}, ptr->data(), freeWhenDone(ptr));
```

Comment:
I find this a bit unsafe, one could go out of bounds easily.
What about passing a single parameter, the one of the width?
```suggestion
py::array asArray(std::vector<T>&& values, ssize_t n_cols) {
    assert values.size() % n_cols == 0,
           "Data mismatch. Values size be a multiple of n_cols"
    const auto ssize_t n_rows = values.size() / n_cols;
    auto ptr = new std::vector<T>(std::move(values));
    return py::array({n_cols, n_rows}, ptr->data(), freeWhenDone(ptr));
```

Reply:
Btw I think I still would prefer using a dedicated type for matrix, other than this `n_cols` workaround